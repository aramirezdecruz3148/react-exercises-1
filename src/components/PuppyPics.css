.puppyPics {
  background: white;
  color: #48D1CC
}