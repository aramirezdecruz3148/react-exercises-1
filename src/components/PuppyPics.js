import React, { useState } from 'react';
import img from '../assets/Pennie.jpg';
import img2 from '../assets/Pennie2.jpg';
import img3 from '../assets/Pennie3.jpg';
import img4 from '../assets/Pennie4.jpg';

const puppyArray = [img, img2, img3, img4];

//make either arrow become disabled 

function PuppyPics() {
  const style = {
    width: '300px',
    height: '300px',
    border: 'solid #FFFF99 5px'
  };

  const [countObj, setCount] = useState({
    count: 0,
    next: puppyArray.length <= 1, //just making a false statement here which will set this value to false
    previous: true
  });

  const incrementNewCount = countObj.count + 1;
  const decrementNewCount = countObj.count - 1;

  const updateState = (updatedCount) => {
    setCount({
      count: updatedCount,
      next: updatedCount >= puppyArray.length - 1,
      previous: updatedCount === 0,
    });
  };

  return (
    <>
      <button disabled={countObj.previous} onClick={() => updateState(decrementNewCount)}>⇦</button>
      <img style={style} src={puppyArray[countObj.count]}/>
      <button disabled={countObj.next} onClick={() => updateState(incrementNewCount)}>⇨</button>
    </>
  );
}

export default PuppyPics;
