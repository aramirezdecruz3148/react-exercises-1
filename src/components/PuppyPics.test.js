import React from 'react';
import { shallow } from 'enzyme';
import PuppyPics from './PuppyPics';

describe('PuppyPics component', () => {
  it('renders PuppyPics', () => {
    const wrapper = shallow(<PuppyPics />);
    expect(wrapper).toMatchSnapshot();
  });
});
