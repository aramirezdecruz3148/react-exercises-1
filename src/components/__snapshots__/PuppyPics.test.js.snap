// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PuppyPics component renders PuppyPics 1`] = `
<Fragment>
  <button
    className="puppyPics"
    disabled={true}
    onClick={[Function]}
  >
    ⇦
  </button>
  <img
    src="/path/image"
    style={
      Object {
        "border": "solid #FFFF99 5px",
        "height": "300px",
        "width": "300px",
      }
    }
  />
  <button
    disabled={false}
    onClick={[Function]}
    style={
      Object {
        "color": "#48D1CC",
      }
    }
  >
    ⇨
  </button>
</Fragment>
`;
